algorithms/spot_finding/finder.py

```diff
@@ -821,7 +821,10 @@ def _find_spots_in_imageset(self, imageset):
 
             logger.info("\nFinding spots in image %s to %s...", j0, j1)
             j0 -= 1
-            r, h = extract_spots(imageset[j0:j1])
+            if len(imageset) == 1:
+                r, h = extract_spots(imageset)
+            else:
+                r, h = extract_spots(imageset[j0:j1])
```

Comment:
As above - it would probably be instructive to understand _why_ this results in an extra image read.

Reply:
See above, but in this case it's because of the extra radial_average step we do in the XFEL GUI.  That causes a data read, so when spotfinding happens and the imageset is sliced, the cache is lost.  In this changeset, the same imageset is used at each step, preventing the cache loss.